frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts

```diff
@@ -0,0 +1,159 @@
+import type { EChartsOption, RegisteredSeriesOption } from "echarts";
+import type { SeriesLabelOption } from "echarts/types/src/util/types";
+import type {
+  SeriesModel,
+  CartesianChartModel,
+} from "metabase/visualizations/echarts/cartesian/model/types";
+import type {
+  ComputedVisualizationSettings,
+  RenderingContext,
+} from "metabase/visualizations/types";
+import type { SeriesSettings } from "metabase-types/api";
+import { isNotNull } from "metabase/lib/types";
+
+const buildEChartsLabelOptions = (
+  seriesModel: SeriesModel,
+  settings: ComputedVisualizationSettings,
+  { getColor, fontFamily, formatValue }: RenderingContext,
+): SeriesLabelOption => {
+  const valueFormatter = (value: unknown) =>
+    formatValue(value, {
+      ...(settings.column?.(seriesModel.column) ?? {}),
+      jsx: false,
+      compact: settings["graph.label_value_formatting"] === "compact",
+    });
+
+  return {
+    show: settings["graph.show_values"],
+    position: "top",
+    fontFamily,
+    fontWeight: 900,
+    fontSize: 12,
+    color: getColor("text-dark"),
+    textBorderColor: getColor("white"),
+    textBorderWidth: 3,
+    formatter: datum => {
+      const dimensionIndex = datum?.encode?.y[0];
+      const dimensionName =
+        dimensionIndex != null ? datum?.dimensionNames?.[dimensionIndex] : null;
+      if (dimensionIndex == null || dimensionName == null) {
```

Comment:
```suggestion
      if (dimensionName == null) {
```

If `dimensionIndex` is `null` or `undefined` then `dimensionName` will be, because of the logic in the previous line, so this check is redundant.